### PR TITLE
storage: Validate fsys label length

### PR DESCRIPTION
**Pull request by mvollmer**
Screenshot for release notes:
xfs-label-too-long

### CODE REVIEW

File: pkg/storaged/utils.js [hunk 1]
```diff
@@ -132,6 +132,26 @@
         }
     };
 
+    utils.validate_fsys_label = function validate_fsys_label(label, type) {
+        var fs_label_max = {
+            "xfs":   12,
+            "ext4":  16,
+            "vfat":  11,
+            "ntfs": 128,
+        };
```
Comment (martinpitt) [inline pkg/storaged/utils.js]: I originally thought [udisks](https://github.com/storaged-project/udisks/blob/master/src/udiskslinuxfsinfo.c) would have a list of these label lengths, but it doesn't. So we even couldn't re-use some validation function from it.

File: test/verify/check-storage-format [hunk 2]
```diff
@@ -67,9 +67,16 @@ class TestStorage(StorageCase):
...
 
-        def check_type(type):
+        def check_type(type, label_limit):
             self.content_tab_action(1, 1, "Format")
-            self.dialog({"type": type})
+            self.dialog_wait_open()
+            self.dialog_set_val("type", type)
+            self.dialog_set_val("name", "X" * (label_limit+1))
+            self.dialog_apply()
+            self.dialog_wait_error("name", "Name cannot be longer than")
```
Comment (martinpitt) [inline test/verify/check-storage-format]: You could even include the length in the expected string.

**Comment by martinpitt** [comment]
pyflakes failure:

    test/verify/check-storage-format:74:59: E226 missing whitespace around arithmetic operator

File: pkg/storaged/utils.js (hunk 1 above)
Comment (mvollmer) [reply pkg/storaged/utils.js]: Yes, exactly.

**Comment by mvollmer** [comment]
> To clarify, there are more things that can go wrong with labels (other file systems, some FSes not allowing all UTF-8 characters, etc.). But for these cases we'd still get and show the udisks error, right?

Yes.

**Comment by mvollmer** [comment]
> pyflakes failure:

Fixed.

**Comment by martinpitt** [comment]
Aiming at tomorrow's release, adding priority.

**Pull request merged**